src/receive_imf.rs

```diff
@@ -601,7 +601,12 @@ pub async fn from_field_to_contact_id(
     prevent_rename: bool,
 ) -> Result<Option<(ContactId, bool, Origin)>> {
     let display_name = if prevent_rename {
-        Some("")
+        Some(
+            match Contact::lookup_id_by_addr(context, &from.addr, Origin::Unknown).await? {
```

Comment:
Not very good is that this adds a new db query, another option is to add a new `Origin` variant right before `IncomingUnknownFrom` so that the fallback display name doesn't overwrite the existing one